### PR TITLE
Replace non-working plot_loss_callback example (#4116)

### PR DESCRIPTION
**Pull request by melentye**
As it says in the #4116 description, current example throws an error because of the dimensions mismatch between np.arange(...) and logs['loss']. One way to fix this is to keep track of all the losses.

### CODE REVIEW

File: keras/callbacks.py [hunk 1]
```diff
@@ -1003,9 +1003,13 @@ class LambdaCallback(Callback):
...
         import numpy as np
         import matplotlib.pyplot as plt
-        plot_loss_callback = LambdaCallback(
-            on_epoch_end=lambda epoch, logs: plt.plot(np.arange(epoch),
-                                                      logs['loss']))
+
+        def plot_loss(epoch, logs, losses=[]):
+            losses.append(logs['loss'])
+            plt.plot(np.arange(epoch + 1), losses)
+            plt.show()  # Use the best method to update the plot for your backend
```
Comment (fchollet) [inline keras/callbacks.py]: Can we not just add points to the graph one at a time?

Comment (melentye) [inline keras/callbacks.py]: Good point, let me see what are the options

Comment (melentye) [inline keras/callbacks.py]: @fchollet the proposed fix is actually creating a plot per epoch which is somewhat suboptimal but better than what we had before 

the shortest code snippet I could come up with that uses an interactive matplotlib backend and can be updated with new data looks like this:

```Python
import numpy as np
import matplotlib as mpl

# Use an interactive backend to get continous updates.
matplotlib.use('TkAgg') 
import matplotlib.pyplot as plt

plt.ion()
fig, ax = plt.subplots()
ax.set_xlim(0, num_epochs)
ax.set_ylim(0, 10)
loss_line, = ax.plot([], [])

def plot_loss(epoch, logs):
    loss_line.set_xdata(np.append(loss_line.get_xdata(), epoch))
    loss_line.set_ydata(np.append(loss_line.get_ydata(), logs['loss']))
    fig.canvas.draw_idle()
    fig.canvas.flush_events()

plot_loss_callback = LambdaCallback(on_epoch_end=plot_loss)

...
```

IMO it's a little too much code for a lambda but what do you think?

Comment (fchollet) [inline keras/callbacks.py]: Then we should use a  different example, like appending to a CSV file. Something that fits in a lambda.

Comment (melentye) [inline keras/callbacks.py]: There's a keras.callbacks.CSVLogger already and it's hard to compete with that. I've updated the example to log JSON lines to a file.

File: keras/callbacks.py [hunk 2]
```diff
@@ -1000,12 +1000,15 @@ class LambdaCallback(Callback):
...
-        plot_loss_callback = LambdaCallback(
-            on_epoch_end=lambda epoch, logs: plt.plot(np.arange(epoch),
-                                                      logs['loss']))
+        # Stream the epoch loss to a file in JSON format. The file content
+        # is not well-formed JSON but rather has a JSON object per line.
+        import json
+        json_log = open('loss_log.json', mode='wt', buffering=1)
+        json_logging_callback = LambdaCallback(
+            on_epoch_end=lambda epoch, logs: json_log.write(
+                json.dumps({'epoch': epoch, 'loss': logs['loss']}) + '\\n'),
```
Comment (fchollet) [inline keras/callbacks.py]: Are you sure about the `\\n`? Why not `\n`?

Comment (melentye) [inline keras/callbacks.py]: I went with `\\n` because it made it render correctly on the website, otherwise it was an actual new line in the docs page. Looking deeper into it, a cleaner solution is to use `r"""raw triple double qoutes"""` for that docstring (see https://www.python.org/dev/peps/pep-0257/ )

**Comment by fchollet** [comment]
LGTM
